### ISSUE TITLE
Menubar breaks cross platform menubar behavior conventions

### ISSUE THREAD

**Issue by Permik** [microsoft/microsoft-ui-xaml#8953]
### Which version of the app?

WinUI 2 Gallery, WinUI 3 Gallery

### Description

Across different platforms you can operate menubars and dropdowns using a "drag-click" that is initiated by pressing down the primary mouse key over the button, keeping the primary mouse key pressed while going through the menus and releasing over the currently hovered/drag-clicked/selected action to activate it.[^1]

To boil it down, the steps below doesn't work anymore. This used to work with win32 top level menu items, but the behavior has regressed.

**Steps to Reproduce**
1. Press down on your mouse on a top level menu item (File, Edit, View, Help)
2. Keep holding down the mouse button
3. The top level menu opens
4. You move the cursor over the menu item you want to activate
5. You release the mouse button

Examples of dropdowns and menubars supporting drag clicking:
Win32 menubars: http://sx.permik.xyz/macup/2023-10-07/04-26-55.mp4 (For some reason the Power Toys Mouse Highlighter gets hidden by the dialogs at ~27 secs)
macOS top level menubar
Android appbar actions, try to swipe downwards on the three vertical dots located on an app bar and keep your finger pressed, release to finish the action.
Web browsers using the  `<select>` element (Safari, Firefox, regressed in Chromium since ???)
Chrome browser (vert_more) and Edge browser (Segoe MDL2/Fluent "more"-icon) top right main dropdown menu

_Yes, this regression also applies to the ComboBox element also, but I'll wait for feedback on this before making an another issue about that._

[^1]: Aside: This behavior works across OS's and is a delightful behavior that feels _really_ good when it's implemented. Something about not needing to click a second time to act on a dropdown that I personally find really satisfying. This also gives the powerusers the ability to "flick" really fast through visual menus because of reducing the need to stop to click something.

### Windows version

Insider Build (xxxxx)

### Additional context

Version 23H2 OS Build (25967.1000)
WinUI3 Gallery Version: 2.2.0.0

**Comment by fdwr** [comment]
Yes! Efficient useability matters, and this bug is annoying in all the extra left clicks it adds during my day (which already average 4000+). I opened Feedback issue on Win11's Notepad today for this functionality regression, which Kenny Guo pointed to WinUI3 as the culprit. Please fix.

**Comment by Permik** [comment]
Could we get a regression tag for this issue, thanks.

**Comment by Permik** [comment]
Bumping as this is probably still regressed.

Is Microsoft hiring?
I'd be open for work to cut down the 1.9k issues which are affecting the UX-side of things.

**Comment by Permik** [comment]
Not currently using a Windows platform to test, but this is most likely still regressed. Could we get a regression tag for this?

**Comment by Permik** [comment]
Went out of my way and set up a VM with the latest WinUI 3 Gallery currently served to me from your storefront, version 2.9.3.0

This regressed behavior can still be replicated.

Also replicable with the top level menu of the modern Notepad application, ver. 11.2604.5.0

The Snipping tool also has regressed behavior on the top right "horizontal_more" button, which should work like a drop-down, and thus should be drag-clickable. (Compare to how Edge works properly in this instance) Snipping tool ver. 11.2602.47.0

Lots of the File Explorer dropdown buttons should be able to be interacted via drag-clicking, but aren't currently. Windows Explorer ver. 10.0.26100.5074

I'll come back and try to replicate these again after my VM has installed all the updates.
